jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -607,6 +611,16 @@ YUI.add('juju-env-api', function(Y) {
 
       // Define the handler reacting to Juju controller login responses.
       var handleResponse = function(bakery, macaroons, cback, data) {
+        // When using JIMM it's possible that the url we have for the model is
+        // not the correct host and we need to fetch the correct path to be
+        // able to connect to the model.
+        if (data.error && data.error === 'redirection required') {
+          this.redirectInfo(resp =>
```

Comment:
As mentioned, I'd let callers do this, this method here already has enough responsibility.